### ISSUE TITLE
Desktop version

### ISSUE THREAD

**Issue by tonypee** [afreakk/ChromeAudioVisualizerExtension#2]
Hi guys, 

Thanks for the awesome work! Just wondering - would it be possible to have the app work as a desktop app - and visualize for all audio - eg. spotify? This is probably hard since you wouldnt have access to the audio stream? Just wondering, love your work otherwise

**Comment by afreakk** [comment]
Thanks :)
Yeah, im not sure, I havent researched how to do that.
I guess we would need some sort of container to put the html and js in, and hopefully that container has some cross platform way of accessing audio.
I know you can pretty easily access audio on linux through pulseaudio, but it would be nice having it cross platform.
Rewriting the whole thing in a different language would also be cool though, as were not constrained to javascript, we could have better performance :)

**Comment by VeGas-Gh0st** [comment]
id love to see this also, this is an amazing plugin thank you

**Comment by cjmh0101** [comment]
Would also love to see this. This feature would allow you to change tracks using spotify/amazon etc but without the party audience seeing you do it.

**Comment by 99zql** [comment]
I agree, maybe getting it to work on the wallpaper, the effects are crazy good for wallpapers

**Comment by afreakk** [comment]
Good news — this is now possible as of v12.2.1 (live on the Chrome Web Store). The extension now has a **Microphone / system audio** capture source. Combined with your OS's audio routing, it visualizes desktop audio from Spotify, YouTube, games — anything that plays through your speakers. No standalone desktop app needed; the browser already has the audio plumbing.

### How it works

1. Open the extension's settings and flip **Capture source** to **Microphone / system audio**. Grant the extension microphone permission if prompted.
2. Route your desktop output into Chromium's audio *input*, using whichever tool your OS provides (one-time setup):
   - **Linux** (PulseAudio/PipeWire): open `pavucontrol` → **Recording** tab → while the visualizer is active, find the Chromium entry (it only appears in this list once the visualizer is capturing) → change its input from your real mic to `Monitor of <your output device>`.
   - **Windows**: Sound control panel → Recording → right-click → **Show Disabled Devices** → enable **Stereo Mix** → set it as Chromium's input device.
   - **macOS**: install [BlackHole](https://github.com/ExistentialAudio/BlackHole) or [Loopback](https://rogueamoeba.com/loopback/), route your system output into it, then set it as Chromium's input.

Per-app routing is possible on Linux too — see the [readme](https://github.com/afreakk/ChromeAudioVisualizerExtension#capture-source) for the null-sink recipe (lets you visualize Spotify only, while other apps stay out).

Feedback welcome — especially on Windows and macOS, since my testing was Linux-only.
